### ISSUE TITLE
Files.CompleteUploadExternal - unexpected behavior

### ISSUE THREAD

**Issue by ohwwh** [soxtoby/SlackNet#221]
Hi, I try to post message, then using thread_ts of this message send commend and image under this message, using GetUploadUrlExternal + CompleteUploadExternal API.

First I tried: 
1. call GetUploadUrlExternal
2. make post request
3. call CompleteUploadExternal: set files, channeldId, threadTs and initialComment
in this way, things don't go with what I expected - commment is gone, image is uploaded under the first message(not as reply), and thread_ts is also posted as string.

but, when I tried until 2(make post request), then I call CompleteUpladExternal API with powershell directly, everything goes well...... 

my own code is below:

```
public async Task<IList<ExternalFileReference>> PostIcon(string channel_id_or_name, byte[] iconBytes, string comment, string message_ts = null)
{
    //return await _slackApiClient.Files.Upload(iconBytes, "png", "icon.png", "icon", comment, message_ts, new[] { channel_id_or_name }
    //).ConfigureAwait(false);

    var uploadResponse = await _slackApiClient.Files.GetUploadUrlExternal("icon.png", iconBytes.Length, "icon").ConfigureAwait(false);
    var uploadUrl = uploadResponse.UploadUrl;
    var fileReference = new ExternalFileReference{ Id = uploadResponse.FileId, Title = "Icon" };

    var httpClient = new HttpClient();
    var content = new ByteArrayContent(iconBytes);
    content.Headers.ContentType = new MediaTypeHeaderValue("image/png");

    var response = await httpClient.PostAsync(uploadUrl, content).ConfigureAwait(false);
    var responseContent = await response.Content.ReadAsStringAsync().ConfigureAwait(false);

     var completeResponse = await _slackApiClient.Files.CompleteUploadExternal(
        files: new ExternalFileReference[] { fileReference },
        channelId: channel_id_or_name,
        threadTs: message_ts,
        initialComment: comment
    ).ConfigureAwait(false);

    return completeResponse;
}
```

**Comment by soxtoby** [comment]
Hi @ohwwh, thanks for raising this. Turns out the `CompleteUploadExternal` implementation had its parameters around the wrong way, so the `initialComment` was being used as the `threadTs` and vice-versa 😬. I'll push out a new release in the next few days with the fix.

That said, you're probably better off using the `Upload` method that takes in a `FileUpload` and handles all these requests for you. Your function would then look like this:

```c#
async Task<ExternalFileReference> PostIcon(string channel_id_or_name, byte[] iconBytes, string comment, string message_ts = null)
{
    return await _slackApiClient.Files.Upload(
            file: new FileUpload("icon.png", iconBytes) { Title = "Icon" },
            channelId: channel_id_or_name,
            threadTs: message_ts,
            initialComment: comment)
        .ConfigureAwait(false);
}
```

This should work as-is, since `Upload` was taking the incorrect parameter ordering into account, which is probably how I missed it.

Hope this helps.

**Comment by soxtoby** [comment]
Should be fixed now in [v0.15.3](https://github.com/soxtoby/SlackNet/releases/tag/v0.15.3).
